test/integration/helpers_test.go

```diff
@@ -281,6 +282,92 @@ func gatewayLinkStatusMatches(
 	return false
 }
 
+func parentStatusContainsProgrammedCondition[T routeParentStatusT](
+	parentStatuses []T, controllerName gatewayv1beta1.GatewayController, expectedStatus metav1.ConditionStatus,
+) bool {
+	var conditions []metav1.Condition
+	parentFound := false
+	for _, parentStatus := range parentStatuses {
+		switch p := (any)(parentStatus).(type) {
+		case gatewayv1beta1.RouteParentStatus:
+			if p.ControllerName == controllerName {
+				conditions = p.Conditions
+				parentFound = true
+			}
+		case gatewayv1alpha2.RouteParentStatus:
+			if gatewayv1beta1.GatewayController(p.ControllerName) == controllerName {
+				conditions = p.Conditions
+				parentFound = true
+			}
+		}
+
+		if parentFound {
+			break
+		}
+	}
+
+	if !parentFound {
+		return false
+	}
+	return lo.ContainsBy(conditions, func(cond metav1.Condition) bool {
+		return cond.Type == "Programmed" && cond.Status == expectedStatus
+	})
+}
+
+func verifyProgrammedConditionStatus(t *testing.T,
+	c *gatewayclient.Clientset,
+	protocolType gatewayv1beta1.ProtocolType,
+	namespace, name string,
+	expectedStatus metav1.ConditionStatus,
+) bool {
+	// gather a fresh copy of the route, given the specific protocol type
+	switch protocolType { //nolint:exhaustive
+	case gatewayv1beta1.HTTPProtocolType:
+		route, err := c.GatewayV1beta1().HTTPRoutes(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			t.Logf("error getting http route: %v", err)
```

Comment:
```suggestion
			t.Logf("error getting http route: %w", err)
```

Reply:
🤔 What would be the benefit of doing this? We're not passing the wrapped error to the caller here but logging it.

Reply:
From the [official doc](https://go.dev/blog/go1.13-errors) here: 
```
In Go 1.13, the fmt.Errorf function supports a new %w verb. When this verb is present, the error returned by fmt.Errorf will have an Unwrap method returning the argument of %w, which must be an error. In all other ways, %w is identical to %v.
```
Seems that there are no differences. 
in existing code, `fmt.Errorf` uses `%w`, but most of logging code uses `%v`. So I would not change here. If you think it is better to use `%w` in all places where we want to format an error, please submit an issue or a PR.

Reply:
sure, let's keep %v :)